services/newsfeed/tests/functional/mysql.test.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const mysqlIndex = require('../../shared/indexdb/mysql');
+const testschema = require('../../shared/schema/newsfeed');
+const {
+	invalidSchemaWithMissingType,
+	invalidSchemaWithoutPrimaryKey,
+} = require('../schemas/invalidSchemas');
+
+const getIndex = () => mysqlIndex('testSchemaNewsfeed', testschema);
+
+const { news, drupalData } = require('../constants/newsfeed');
+
+describe('Test mysql', () => {
+	let db;
+	beforeAll(async () => {
+		// Create table
+		db = await getIndex();
+	});
+
+	afterAll(async () => {
+		// TODO: Drop table
+		const result = await db.find();
+		await db.deleteIds(result.map(r => r[`${testschema.primaryKey}`]));
+	});
+
+	it('Valid schema', async () => {
+		const getValidIndex = () => mysqlIndex('ValidTestSchema', testschema);
+		const testDB = await getValidIndex();
+		expect(testDB).toBeInstanceOf(Object);
+		expect(testDB).toEqual(
+			expect.objectContaining({
+				upsert: expect.any(Function),
+				find: expect.any(Function),
+				count: expect.any(Function),
+				deleteIds: expect.any(Function),
+				rawQuery: expect.any(Function),
+				increment: expect.any(Function),
+			}),
+		);
+	});
+
+	// TODO: Update test case when proper error handling is implemented
+	it('Invalid schema: Missing primary key', async () => {
+		const getInvalidIndex = () => mysqlIndex('TestSchemaWithoutPrimaryKey', invalidSchemaWithoutPrimaryKey);
+		const testDB = await getInvalidIndex();
+		await testDB.upsert(news);
+		expect(testDB.find()).rejects.toThrow();
+	});
+
+	it('Invalid schema: Missing type defined', async () => {
+		const getInvalidIndex = () => mysqlIndex('TestSchemaWithoutType', invalidSchemaWithMissingType);
+		expect(getInvalidIndex()).rejects.toThrow();
+	});
```

Comment:
Please make sure to drop the tables after the tests have run. Those statements can be part of the same test case or better `afterAll()`. Use `IF EXIST` to avoid errors while removing the tables, esp. `TestSchemaWithoutType`